lib/systeminfo/user.go

```diff
@@ -76,43 +73,76 @@ func GetRealUser() (*User, error) {
 
 // LookupUserByName finds a user by name
 func LookupUserByName(name string) (*User, error) {
-	u, err := lookupUser(func(u user.User) bool {
-		return u.Name == name
-	})
+	usr, err := user.Lookup(name)
 	if err != nil {
-		return nil, trace.Wrap(convertError(err.Error()))
+		return nil, trace.Wrap(convertUserError(err))
+	}
+	uid, err := strconv.Atoi(usr.Uid)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	gid, err := strconv.Atoi(usr.Gid)
+	if err != nil {
+		return nil, trace.Wrap(err)
 	}
-
 	return &User{
-		Name: u.Name,
-		UID:  u.Uid,
-		GID:  u.Gid,
+		Name: usr.Username,
+		UID:  uid,
+		GID:  gid,
 	}, nil
 }
 
 // LookupUserByUID finds a user by ID
 func LookupUserByUID(uid int) (*User, error) {
-	u, err := lookupUser(func(u user.User) bool {
-		return u.Uid == uid
-	})
+	usr, err := user.LookupId(strconv.Itoa(uid))
 	if err != nil {
-		return nil, trace.Wrap(convertError(err.Error()))
+		return nil, trace.Wrap(convertUserError(err))
+	}
+	gid, err := strconv.Atoi(usr.Gid)
+	if err != nil {
+		return nil, trace.Wrap(err)
 	}
-
 	return &User{
-		Name: u.Name,
-		UID:  u.Uid,
-		GID:  u.Gid,
+		Name: usr.Username,
+		UID:  uid,
+		GID:  gid,
 	}, nil
 }
 
+func checkGroup(groupName, groupID string) error {
```

Comment:
nit: I would place these in the newspaper order - i.e. after the exported `NewUser` that uses them. I know it's not a widespread style convention in the Go community but I feel it adheres the readability.

Reply:
I agree... I forgot to address it and CI just finished. I'll fix this in another PR.